linalg/sparsemat.cpp

```diff
@@ -1457,10 +1509,21 @@ void SparseMatrix::EliminateRowCol(int rc, SparseMatrix &Ae, int d)
       {
          if ((col = nd->Column) == rc)
          {
-            if (d == 0)
+            switch(dpolicy)
             {
-               Ae.Add(rc, rc, nd->Value - 1.0);
-               nd->Value = 1.0;
+            case DIAG_ONE:
+              Ae.Add(rc, rc, nd->Value - 1.0);
+              nd->Value = 1.0;
+              break;
+            case DIAG_ZERO:
+//              Ae.Add(rc, rc, nd->Value); ???
```

Comment:
I believe this line should be uncommented.  The goal of this routine is that the original matrix A is the sum of the two new matrices.